Sources/lit-test-helper/main.swift

```diff
@@ -371,15 +371,16 @@ func performRoundtrip(args: CommandLineArguments) throws {
   }
 }
 
-class NodePrinter: SyntaxVisitor {
-  override func visitPre(_ node: Syntax) {
+struct NodePrinter: SyntaxAnyVisitor {
```

Comment:
Maybe related question: why do we change these `class` to `struct`?

Reply:
Because protocol-based programming allows us to use value types which make us happy inside ? 😁
I mean changing `SyntaxVisitor` to protocol allowed us to use either structs or classes, so why not take advantage of it and switch to value types.